### PR TITLE
Configure AKS Sku Tier (Uptime SLA) & diagnostic logs options

### PR DESCRIPTION
**Pull request by thomast1906**
### Change description ###
Configure AKS Sku Tier (Uptime SLA) & diagnostic logs options

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: 01-inputs-required.tf [hunk 1]
```diff
@@ -27,3 +27,11 @@ variable "project_acr_enabled" {
 variable "global_acr_enabled" {
   default = true
 }
+
+variable "monitor_diagnostic_setting" {
+  default = false
```
Comment (timja) [inline 01-inputs-required.tf:32]: same comment as https://github.com/hmcts/aks-sds-deploy/pull/72#discussion_r657939078

File: 01-inputs-required.tf [hunk 2]
```diff
@@ -27,3 +27,11 @@ variable "project_acr_enabled" {
...
+}
+
+variable "sku_tier" {
+  default = "Free"
```
Comment (timja) [inline 01-inputs-required.tf:36]: add description with available values? Free or Paid isn't it?

Comment (thomast1906) [reply 01-inputs-required.tf:36]: Yup, will update

File: 01-inputs-required.tf (hunk 1 above)
Comment (thomast1906) [reply 01-inputs-required.tf:32]: Sure - will `true `here

File: 12-kubernetes-cluster.tf [hunk 3]
```diff
@@ -136,3 +138,45 @@ resource "azurerm_kubernetes_cluster_node_pool" "additional_node_pools" {
...
+    enabled  = true
+  }
+
+  log {
+    category = "kube-scheduler"
+    enabled  = true
+  }
+
+  log {
+    category = "kube-audit"
```
Comment (timja) [inline 12-kubernetes-cluster.tf:169]: this is what I was talking about yesterday: https://docs.microsoft.com/en-us/azure/aks/view-control-plane-logs#log-categories

kube-audit isn't really recommended as it has a really high cost
```suggestion
    category = "kube-audit-admin"
```

File: 12-kubernetes-cluster.tf [hunk 4]
```diff
@@ -136,3 +138,45 @@ resource "azurerm_kubernetes_cluster_node_pool" "additional_node_pools" {
...
+    enabled  = true
+  }
+
+  log {
+    category = "cluster-autoscaler"
+    enabled  = true
+  }
+
+  log {
+    category = "kube-scheduler"
```
Comment (timja) [inline 12-kubernetes-cluster.tf:164]: maybe enable the `guard` category as well?

File: 01-inputs-required.tf (hunk 1 above)
Comment (thomast1906) [reply 01-inputs-required.tf]: Default now set as `true`

File: 01-inputs-required.tf (hunk 2 above)
Comment (timja) [reply 01-inputs-required.tf:37]:
```suggestion
  default     = "Free"
  description = "Free or Paid (which includes the uptime SLA)"
```

File: 01-inputs-required.tf [hunk 5]
```diff
@@ -27,3 +27,12 @@ variable "project_acr_enabled" {
 variable "global_acr_enabled" {
   default = true
 }
+
+variable "monitor_diagnostic_setting" {
+  default = true
+}
+
+# Free or Paid (which includes the uptime SLA)
```
Comment (timja) [inline 01-inputs-required.tf:35]:
```suggestion
```

File: 01-inputs-required.tf (hunk 2 above)
Comment (timja) [reply 01-inputs-required.tf:37]: description element it used for generating docs (used in most repoes although apparently not this one)

**Review by timja — APPROVED** [review]

**Pull request merged**